rust/chains/hyperlane-cosmos/src/mailbox.rs

```diff
@@ -141,15 +185,22 @@ impl Mailbox for CosmosMailbox {
     async fn recipient_ism(&self, recipient: H256) -> ChainResult<H256> {
         let address = verify::digest_to_addr(recipient, &self.signer.prefix)?;
 
-        let payload = mailbox::DefaultIsmRequest {
-            default_ism: general::EmptyStruct {},
+        let payload = mailbox::ISMSpecifierRequest {
+            interchain_security_module: vec![],
         };
 
         let data = self.provider.wasm_query_to(address, payload, None).await?;
-        let response: mailbox::DefaultIsmResponse = serde_json::from_slice(&data)?;
+
+        let response: mailbox::ISMSpecifierResponse = serde_json::from_slice(&data)?;
 
         // convert Hex to H256
-        let ism = H256::from_slice(&hex::decode(response.default_ism)?);
+        let default_ism = self.default_ism().await?;
```

Comment:
the recipient ISM function should handle the default fallback at the contract level